agenda-services/src/main/java/org/exoplatform/agenda/rest/AgendaEventRest.java

```diff
@@ -1713,4 +1715,75 @@ public Response getPendingEvents(
     }
   }
 
+
+  @Path("ics")
+  @GET
+  @Produces(MediaType.TEXT_PLAIN)
+  @RolesAllowed("users")
+  @Operation(
+          summary = "Returns the ICS file generated from this event",
+          description = "Returns the ICS file generated from this event.",
+          method = "GET"
+  )
+  @ApiResponses(
+          value = {
+                  @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+                  @ApiResponse(responseCode = "400", description = "Invalid query parameters"),
+                  @ApiResponse(responseCode = "500", description = "Internal server error"),
+          }
+  )
+  public Response getICSOfEvent(
+          @Context
+          HttpServletRequest request,
+          @Parameter(description= "The event Id")
+          @QueryParam(
+                  "eventId"
+          )
+          long eventId,
+          @Parameter(description= "the timezone ID of the target user")
+          @QueryParam(
+                  "timeZoneId"
+          )
+          String timeZoneId) {
+    try {
+      if (eventId <= 0) {
+        return Response.status(Status.BAD_REQUEST).entity("the eventId param is required").build();
+      }
+      if (StringUtils.isBlank(timeZoneId)) {
+        timeZoneId = TimeZone.getDefault().getID();
+      }
+      Identity currentUserIdentity = this.identityManager.getOrCreateUserIdentity(request.getRemoteUser());
```

Comment:
Should it be better to use the new way to get current user : 
https://community.exoplatform.com/portal/dw/activity?id=445564
?

Reply:
@rdenarie Here I am getting just the username of the current user from the HTTP request and wants to check if the user exists, and did not use the user ACL since I do not check permissions. Could you give more details about your suggestion ?

Reply:
My bad, I misunderstood the content of [community.exoplatform.com/portal/dw/activity?id=445564](https://community.exoplatform.com/portal/dw/activity?id=445564)

What is better between ConversationState.getCurrent and request.getRemoteUser ?

Reply:
@rdenarie IMO request.getRemoteUser is better as it is based on the HTTP servlets  standard. ConversationState.getCurrent is based on ThreadLocal which is a specific implementation of eXo core, and may need to change when we will go stateless .